idaes/models/properties/modular_properties/reactions/equilibrium_forms.py

```diff
@@ -287,7 +287,8 @@ def return_expression(b, rblock, r_idx, T):
         Q = b.log_k_eq[r_idx] - e
         # Q should be unitless due to log form
 
-        return s - smooth_max(0, s - Q, rblock.eps) == 0
+        s = s * 10 / (s + b.k_eq[r_idx])
```

Comment:
What are typical values for k_eq. I am thinking about the scaling of s with the proposed reformulation.

When k_eq = 0, s <-- 10

When k_eq = 10,000, s <-- s / 1,000 (approximately)

Is this the intended behavior?

Comments could help explain on of @andrewlee94's questions better.

Reply:
Besides the comments above, to compare `s` and `Q` (decide which one is the zero term), we also need similar sensitivity of both terms at low concentrations. `Q` in log form is very sensitive to low concentrations and the first derivative of Q, dQ=1/f(c).

The first derivative of the updated `s` term is dS=Ksp/(s+Ksp)^2, for s << Ksp (low precipitate), dS=1/Ksp, Ksp>=f(c), so I consider a factor that also matches the sensitivity and 10 works best.

Reply:
OK, I think I understand where the 10 is coming from, but that means I think it should be a mutable param as well rather than a hard-coded value so that users can tune this parameter if required.

Reply:
> Q in log form is very sensitive to low concentrations and the first derivative of Q, dQ=1/f(c).

Could you go more into the sensitivity to low concentrations? It looks like it's using the log concentration forms like it should.

Reply:
> > Q in log form is very sensitive to low concentrations and the first derivative of Q, dQ=1/f(c).
> 
> Could you go more into the sensitivity to low concentrations? It looks like it's using the log concentration forms like it should.

Yes, this is an improved form proposed for `Q` in log form at the beginning. As you may know, the log transform of the equilibrium reaction would perform better at low concentrations. It also improves the sensitivity of Q over component concentrations for solubility products. 

The constraint here is to decide which is the "exact" zero term between `Q` and `S`. So for those low-concentration cases (e.g., Ksp ~1e-20), the reformated precipitate term S=s/(s+Ksp) is going 1 when `s` is relatively large compared to Ksp or 0 when `s` is relatively small compared to Ksp. `s` in this way is able to keep a similar precision as f(c). This idea also works for the non-log-transformed form.

Besides a similar precision, we also prefer a similar sensitivity of `Q` and `S` over state variables in this constraint. The first derivatives in previous comments are evaluated for dQ/df(c) and dS/ds for log-transformed form. This is the form we found better performance in some low concentration cases from WaterTAP.